public/app/core/components/TimelineChart/utils.ts

```diff
@@ -445,15 +446,61 @@ export function prepareTimelineFields(
   const frames: DataFrame[] = [];
 
   for (let frame of series) {
-    let isTimeseries = false;
+    let startFieldIdx = -1;
+    let endFieldIdx = -1;
+
+    for (let i = 0; i < frame.fields.length; i++) {
+      let f = frame.fields[i];
+
+      if (f.type === FieldType.time) {
+        if (startFieldIdx === -1) {
+          startFieldIdx = i;
+        } else if (endFieldIdx === -1) {
+          endFieldIdx = i;
+          break;
+        }
```

Comment:
I remember that we have handling were we have consider the first field to be the time field, does it make sense though to always consider the next found time field as the end field? (i.e. couldn't other time data be represented in user's data?)

I'm not sure this should be default behavior vs only apply in certain circumstances 🤷🏻‍♂️

Reply:
> i.e. couldn't other time data be represented in user's data?

technically, yes. but there is plenty of existing "auto" behavior that makes assumptions like "first string field", "all number fields", etc. to get it perfect in all cases we'd have to give explicit control to users over what each field in their response represents for each viz panel.

my gut feeling is that endTime is the most likely additional time field. if this turns out to be wrong, we can then expose options to override these assumptions and select fields explicitly, like we do in Trend, Candlestick, XYChart, etc.